server/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -234,16 +235,27 @@ static SecureSettings loadSecureSettings(Environment initialEnv) throws Bootstra
             throw new BootstrapException(e);
         }
 
+        char[] password;
+        if (keystore != null && keystore.hasPassword()) {
+            password = Terminal.DEFAULT.readSecret("Elasticsearch password? ");
+        } else {
+            // TODO[wrb]: is there a case w/o stdin?
+            password = new char[0];
+        }
+
         try {
             if (keystore == null) {
                 final KeyStoreWrapper keyStoreWrapper = KeyStoreWrapper.create();
                 keyStoreWrapper.save(initialEnv.configFile(), new char[0]);
                 return keyStoreWrapper;
             } else {
-                keystore.decrypt(new char[0] /* TODO: read password from stdin */);
-                KeyStoreWrapper.upgrade(keystore, initialEnv.configFile(), new char[0]);
+                keystore.decrypt(password);
+                KeyStoreWrapper.upgrade(keystore, initialEnv.configFile(), password);
             }
         } catch (Exception e) {
+            if (password.length != 0) {
+                throw new BootstrapException(new SecurityException("Incorrect keystore password"));
+            }
```

Comment:
Add a `finally` clause where we would call `close()` on the SecureString with the password

Reply:
I'm going to use a try-with-resources and let Java take care of it.